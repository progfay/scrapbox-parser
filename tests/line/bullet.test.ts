/* global describe it expect */

describe('bullet', () => {
  it('Single-byte space indent', () => {
    expect(' Single-byte space').toEqualWhenParsing([
      {
        indent: 1,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Single-byte space'
          }
        ]
      }
    ])
  })

  it('Double-byte space indent', () => {
    expect('ã€€Double-byte space').toEqualWhenParsing([
      {
        indent: 1,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Double-byte space'
          }
        ]
      }
    ])
  })

  it('Tab indent', () => {
    // eslint-disable-next-line no-tabs
    expect('	Tab').toEqualWhenParsing([
      {
        indent: 1,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Tab'
          }
        ]
      }
    ])
  })

  it('Multi lines bullet', () => {
    expect(`no bullet (indent: 0)
 first bullet (indent: 1)
  second bullet (indent: 2)
   third bullet (indent: 3)`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'no bullet (indent: 0)'
          }
        ]
      },
      {
        indent: 1,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'first bullet (indent: 1)'
          }
        ]
      },
      {
        indent: 2,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'second bullet (indent: 2)'
          }
        ]
      },
      {
        indent: 3,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'third bullet (indent: 3)'
          }
        ]
      }
    ])
  })
})
