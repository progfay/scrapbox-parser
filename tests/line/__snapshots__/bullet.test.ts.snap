// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bullet > Double-byte space indent 1`] = `
[
  {
    "indent": 1,
    "nodes": [
      {
        "raw": "Double-byte space",
        "text": "Double-byte space",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`bullet > Multi lines bullet 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "no bullet (indent: 0)",
        "text": "no bullet (indent: 0)",
        "type": "plain",
      },
    ],
    "type": "line",
  },
  {
    "indent": 1,
    "nodes": [
      {
        "raw": "first bullet (indent: 1)",
        "text": "first bullet (indent: 1)",
        "type": "plain",
      },
    ],
    "type": "line",
  },
  {
    "indent": 2,
    "nodes": [
      {
        "raw": "second bullet (indent: 2)",
        "text": "second bullet (indent: 2)",
        "type": "plain",
      },
    ],
    "type": "line",
  },
  {
    "indent": 3,
    "nodes": [
      {
        "raw": "third bullet (indent: 3)",
        "text": "third bullet (indent: 3)",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`bullet > Single-byte space indent 1`] = `
[
  {
    "indent": 1,
    "nodes": [
      {
        "raw": "Single-byte space",
        "text": "Single-byte space",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`bullet > Tab indent 1`] = `
[
  {
    "indent": 1,
    "nodes": [
      {
        "raw": "Tab",
        "text": "Tab",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;
